modules/gltf/gltf_document.cpp

```diff
@@ -6894,32 +6910,54 @@ Node *GLTFDocument::generate_scene(Ref<GLTFState> state, int32_t p_bake_fps) {
 			_import_animation(state, ap, i, p_bake_fps);
 		}
 	}
-
+	for (Map<GLTFNodeIndex, Node *>::Element *node_i = state->scene_nodes.front(); node_i; node_i = node_i->next()) {
+		Node *node = state->scene_nodes[node_i->key()];
+		ERR_CONTINUE(!node);
+		for (int32_t ext_i = 0; ext_i < document_extensions.size(); ext_i++) {
+			Ref<GLTFDocumentExtension> ext = document_extensions[ext_i];
+			ERR_CONTINUE(ext.is_null());
+			ERR_CONTINUE(!state->json.has("nodes"));
+			Array nodes = state->json["nodes"];
+			ERR_CONTINUE(node_i->key() >= nodes.size());
+			Dictionary node_json = nodes[node_i->key()];
+			Ref<GLTFNode> gltf_node = state->nodes[node_i->key()];
+			err = ext->import_node(ext->get_import_options(), state, gltf_node, node_json, node);
+			ERR_CONTINUE(err != OK);
+		}
+	}
 	for (int32_t ext_i = 0; ext_i < document_extensions.size(); ext_i++) {
 		Ref<GLTFDocumentExtension> ext = document_extensions[ext_i];
 		ERR_CONTINUE(ext.is_null());
-		Error err = ext->import_post(this, root);
-		ERR_FAIL_COND_V(err != OK, nullptr);
+		err = ext->import_post(ext->get_import_options(), state, root);
+		ERR_CONTINUE(err != OK);
 	}
 	ERR_FAIL_NULL_V(root, nullptr);
 	return root;
 }
 
-Error GLTFDocument::append_from_scene(Node *p_node, Ref<GLTFState> state, uint32_t p_flags, int32_t p_bake_fps) {
-	ERR_FAIL_COND_V(state.is_null(), FAILED);
-	state->use_named_skin_binds =
+Error GLTFDocument::append_from_scene(Node *p_node, Ref<GLTFState> r_state, uint32_t p_flags, int32_t p_bake_fps) {
+	ERR_FAIL_COND_V(r_state.is_null(), FAILED);
+	r_state->use_named_skin_binds =
 			p_flags & GLTF_IMPORT_USE_NAMED_SKIN_BINDS;
-
-	_convert_scene_node(state, p_node, -1, -1);
-	if (!state->buffers.size()) {
-		state->buffers.push_back(Vector<uint8_t>());
+	Error err = OK;
+	for (int32_t ext_i = 0; ext_i < document_extensions.size(); ext_i++) {
+		Ref<GLTFDocumentExtension> ext = document_extensions[ext_i];
+		ERR_CONTINUE(ext.is_null());
+	}
+	// Security Note:
+	// GLTFDocument extensions has the executing process's permission to read accessible files because guided steps are
+	// a possible feature.
```

Comment:
I find this note confusing. Is this a special security consideration beyond what GDScript is already capable of?

If the security guarantees are the same as GDScript, I don't think we should have this note here.

What are "guided steps"?